modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -299,6 +299,22 @@ class SeqexecEngine(httpClient: Client[IO], settings: SeqexecEngine.Settings, sm
     }}.map(_ => StartQueue(qid, clientId)))
   ).map(_.asRight)
 
+  private def stopSequencesInQueue(qid: QueueId): executeEngine.HandleType[Unit] =
+    executeEngine.get.map(st => (EngineState.queues ^|-? index(qid)).getOption(st).map(_.queue.filter(sid => (EngineState.executionState ^|-> Engine.State.sequences ^|-? index(sid)).getOption(st).map(_.status.isRunning).getOrElse(false))).getOrElse(List.empty)).flatMap(_.map(executeEngine.pause).fold(executeEngine.unit)(_ *> _))
```

Comment:
Could you reorg this line a bit, to make it a bit more vertical
You can also replace `map().getOrElse(Nil)` with `foldMap` I think